src/dmd/cparse.d

```diff
@@ -1442,6 +1443,30 @@ final class CParser(AST) : Parser!AST
                 }
             }
 
+            /* GNU Extensions
+             * init-declarator:
+             *    declarator simple-asm-expr[opt] gnu-attributes[opt]
+             *    declarator simple-asm-expr[opt] gnu-attributes[opt] = initializer
+             */
+            switch (token.value)
+            {
+                case TOK.assign:
+                case TOK.comma:
+                case TOK.semicolon:
+                case TOK.asm_:
+                case TOK.__attribute__:
+                    /* This is a data definition, there cannot now be a
+                     * function definition.
+                     */
+                    first = false;
+                    if (token.value == TOK.asm_)
+                        asmname = cparseAsmLabel();
```

Comment:
Is cparseAsmLabel parsing simple-asm-expr or gnu-attributes? The name should reflect the grammar?

Reply:
#12551